src/jvm/main/org/jetbrains/kotlinx/lincheck/LincheckOptions.kt

```diff
@@ -0,0 +1,394 @@
+        _customScenarios.add(scenario)
+    }
+
+    /**
+     * Examine the specified custom scenario additionally to the generated ones.
+     */
+    fun addCustomScenario(scenarioBuilder: DSLScenarioBuilder.() -> Unit) =
+        addCustomScenario(scenario { scenarioBuilder() })
+
+    /**
+     * If this feature is enabled and an invalid interleaving has been found,
+     * *lincheck* tries to minimize the corresponding scenario in order to
+     * construct a smaller one so that the test fails on it as well.
+     * Enabled by default.
+     */
+    fun minimizeFailedScenario(minimizeFailedScenario: Boolean) = apply {
+        this.minimizeFailedScenario = minimizeFailedScenario
+    }
+
+    /**
+     * The mode used for running tests.
+     *
+     * @see LincheckMode
+     */
+    fun mode(mode: LincheckMode) = apply {
+        this.mode = mode
+    }
+
+    /**
+     * Number of different test scenarios to be executed.
+     */
+    fun iterations(iterations: Int) = apply {
+        this.iterations = iterations
+        this.adjustIterations = false
+    }
+
+    /**
+     * Run each test scenario the specified number of times.
+     */
+    fun invocationsPerIteration(invocations: Int) = apply {
+        this.invocations = invocations
+        this.adjustInvocations = false
+    }
+
+    /**
+     * The maximal amount of time dedicated to testing.
+     */
+    internal fun testingTime(timeMs: Long) = apply {
+        this.testingTimeMs = timeMs
+    }
+
+    /**
+     * Internal, DO NOT USE.
+     */
+    internal fun invocationTimeout(timeoutMs: Long) = apply {
+        this.invocationTimeoutMs = timeoutMs
+    }
+
+    /**
+     * Use the specified verifier.
+     */
+    fun verifier(verifier: Class<out Verifier?>) = apply {
+        this.verifier = verifier
+    }
+
+    /**
+     * The specified class defines the sequential behavior of the testing data structure;
+     * it is used by [Verifier] to build a labeled transition system,
+     * and should have the same methods as the testing data structure.
+     *
+     * By default, the provided concurrent implementation is used in a sequential way.
+     */
+    fun sequentialSpecification(clazz: Class<*>?) = apply {
+        sequentialSpecification = clazz
+    }
+
+    /**
+     * Add a guarantee that methods in some classes are either correct in terms of concurrent execution or irrelevant.
+     * These guarantees can be used for optimization. For example, we can add a guarantee that all the methods
+     * in `java.util.concurrent.ConcurrentHashMap` are correct and this way the strategy will not try to switch threads
+     * inside these methods. We can also mark methods irrelevant (e.g., in logging classes) so that they will be
+     * completely ignored (so that they will neither be treated as atomic nor interrupted in the middle) while
+     * studying possible interleavings.
+     */
+    fun addGuarantee(guarantee: ManagedStrategyGuarantee) = apply {
```

Comment:
Nobody uses this feature, and it is hard to maintain. Let's remove it from the API.